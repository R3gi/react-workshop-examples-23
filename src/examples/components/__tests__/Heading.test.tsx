import { render } from '@testing-library/react';
import Heading, { TUser } from '../Heading';

describe('<Heading />', () => {
  describe('#render', () => {
    const user: TUser = {
      name: 'foobar',
    };
    it('should render component with default props', () => {
      const { container } = render(<Heading user={user}>text</Heading>);

      expect(container).toMatchSnapshot();
    });

    it('should render component with title', () => {
      const { container } = render(
        <Heading user={user} title="bar">
          text
        </Heading>,
      );

      expect(container).toMatchSnapshot();
    });
  });
});
