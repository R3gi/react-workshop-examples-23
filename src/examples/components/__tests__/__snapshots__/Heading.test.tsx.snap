// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> #render should render component with { title: 'bar' } 1`] = `
<div>
  <h1>
    bar
  </h1>
  text
  <p>
    foobar
  </p>
  <button
    type="button"
  >
    btn
  </button>
</div>
`;

exports[`<Heading /> #render should render component with { title: undefined } 1`] = `
<div>
  <h1>
    foo
  </h1>
  text
  <p>
    foobar
  </p>
  <button
    type="button"
  >
    btn
  </button>
</div>
`;

exports[`<Heading /> #render should render component with default props 1`] = `
<div>
  <h1>
    foo
  </h1>
  text
  <p>
    foobar
  </p>
  <button
    type="button"
  >
    btn
  </button>
</div>
`;

exports[`<Heading /> #render should render component with title 1`] = `
<div>
  <h1>
    bar
  </h1>
  text
  <p>
    foobar
  </p>
  <button
    type="button"
  >
    btn
  </button>
</div>
`;
